core/templates/services/user-backend-api.service.ts

```diff
@@ -27,21 +27,32 @@ export interface SubscriptionSummary {
   'creator_impact': number;
 }
 
-export interface PreferencesBackendDict {
+export interface EmailPreferencesBackendDict {
+  'can_receive_email_updates': boolean;
+  'can_receive_editor_role_email': boolean;
+  'can_receive_feedback_message_email': boolean;
+  'can_receive_subscription_email': boolean;
+}
+
+interface NonEmailPreferencesBackendDict {
   'preferred_language_codes': string[];
   'preferred_site_language_code': string;
   'preferred_audio_language_code': string;
   'profile_picture_data_url': string;
   'default_dashboard': string;
   'user_bio': string;
   'subject_interests': string;
-  'can_receive_email_updates': boolean;
-  'can_receive_editor_role_email': boolean;
-  'can_receive_feedback_message_email': boolean;
-  'can_receive_subscription_email': boolean;
   'subscription_list': SubscriptionSummary[];
 }
 
+// The following type is an intersection of EmailPreferencesBackendDict
```

Comment:
intersection ≠ union

Reply:
Yes, this type is **not** a union of those two interfaces. union implies that it could be either `EmailPreferencesBackendDict` or `NonEmailPreferencesBackendDict`. But in our case, the type has to include the properties of **both** the interfaces and not just one and thats why an intersection was used.
https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#intersection-types

Reply:
yes, the set theory does imply a different meaning. updated comment!